test/runSaga.js

```diff
@@ -67,5 +63,13 @@ test('runSaga', assert => {
   )
 
   task.done.catch(err => assert.fail(err))
-
 })
+
+test('put causing sync dispatch response in store-like subscriber', assert => {
+  const reducer = (state, action) => action.type
+  const store = storeLike(reducer, {})
+
+  const { rootSaga, runTest } = syncDispatchTest(assert, store)
```

Comment:
wouldn't be simpler to pass `runSaga` as argument to `syncDispatchTest`. it can then run sagas and tests directly